### PR TITLE
L-585: Add support for Clio regions other than NA

### PR DESCRIPTION
**Pull request by amackera**
Introducing support for **Clio Regions**!

Clio API supports four geographic regions: 

| Region Code | Geographic Area | API |
| ------------- | ----------------- | -----|
|`US` | United States + Territories | `app.clio.com` |
| `CA` | Canada | `ca.app.clio.com` |
| `EU` | European Union (EMEA) | `eu.app.clio.com` |
| `AU` | Australia | `au.app.clio.com` |

## Region support in `hyacinth`
`hyacinth` defaults to the `US` region:
```python
>>> import hyacinth
>>> session = hyacinth.Session(
        token=<token>, 
        client_id=<client_id>,
        client_secret=<client_secret>
)
>>> session.make_url("users")
'https://app.clio.com/api/v4/users.json'
```

To enable region support, supply `region` parameter, which must be one of `['US', 'CA', 'EU', 'AU']`.
```python
>>> ca_session = hyacinth.Session(
        token=<token>, 
        client_id=<client_id>,
        client_secret=<client_secret>,
        region="CA"
)
>>> ca_session.make_url("users")
'https://ca.app.clio.com/api/v4/users.json'
```

### Other Notes
- I've added a few simple tests for region support as well, I'd love some thoughts on what other ways I can test this!

### CODE REVIEW

File: hyacinth/session.py [hunk 1]
```diff
@@ -54,29 +63,50 @@ def wrapper(self, *args, **kwargs):
...
+            self.base_url = CLIO_BASE_URL_CA
+            self.api_base_url = CLIO_API_BASE_URL_CA
+        elif region == "AU":
+            self.base_url = CLIO_BASE_URL_AU
+            self.api_base_url = CLIO_API_BASE_URL_AU
+        elif region == "EU":
+            self.base_url = CLIO_BASE_URL_EU
+            self.api_base_url = CLIO_API_BASE_URL_EU
+        else:
+            log.warning(f"Invalid region supplied: {region}, defaulting to 'US'")
```
Comment (nfranchise) [inline hyacinth/session.py:98]: Do we want to default to the US or just throw an error and leave it at that? If the user & account in question is on a different geo instance, it'll fail if we try to make a request with their token on the US instance. Right?

File: hyacinth/session.py [hunk 2]
```diff
@@ -85,11 +115,11 @@ def __init__(
         self.ratelimit_remaining = math.inf
         self.raise_for_status = raise_for_status
 
-    @staticmethod
-    def __make_url(path):
-        return f"{CLIO_API_BASE_URL_US}/{path}.json"
+    def make_url(self, path):
```
Comment (nfranchise) [inline hyacinth/session.py:118]: Curious why you made this change?

File: test/test_session.py
Comment (nfranchise) [inline test/test_session.py:1]: Beautiful work here 🙌

**Review by nfranchise — CHANGES_REQUESTED** [review]
Looks funky fresh! Two questions for ya

File: hyacinth/session.py (hunk 2 above)
Comment (amackera) [reply hyacinth/session.py:118]: This one was to make it easier to test the changes. I also thought to myself "Why make this private? 🤔 "

File: hyacinth/session.py (hunk 1 above)
Comment (amackera) [reply hyacinth/session.py:98]: This is if they provide a region like "JP" or "CN", one that Clio has no regional support for

Comment (amackera) [reply hyacinth/session.py:98]: If the token doesn't make the provided region, this session's requests will fail

Comment (nfranchise) [reply hyacinth/session.py:98]: Ok sounds good. Makes sense since `US` is the default regardless

**Review by nfranchise — APPROVED** [review]
🥳 Excited to see these get smerged in!

**Comment by amackera** [comment]
Thanks for the review my good sir

**Pull request merged**